__tests__/app.test.js

```diff
@@ -23,4 +23,58 @@ describe('GET /api', () => {
           expect(body.endpoints).toEqual(endpoints);
         });
     });
-  });
\ No newline at end of file
+  });
+
+  describe("invalid endpoint", () => {
+    test("responds with a 404 status and an arror message ehen given an endpoint that does nor exist", () => {
+      return request(app)
+      .get("/api/not-an-endpoint")
+      .expect(404)
+      .then(({body}) => {
+        expect(body.message).toBe('Not Found')
+      })
+    })
+  })
+
+  describe('GET /api/topics', () => {
+    test('responds with status 200 and an array containing data of all topics', () => {
+      return request(app)
+        .get("/api/topics")
+        .expect(200)
+        .then(({ body }) => {
+          expect(Array.isArray(body.topics)).toBe(true);
+          expect(body.topics.length).toBeGreaterThan(0);
+  
+          expect(body.topics[0]).toEqual({
+            description: 'The man, the Mitch, the legend',
+            slug: 'mitch'
+          });
+  
+          body.topics.forEach(topic => {
+            expect(topic).toEqual(
+              expect.objectContaining({
+                slug: expect.any(String),
+                description: expect.any(String)
+              })
+            );
+          });
+        });
```

Comment:
This test is good, and very robust! 
I think you could have used less code, to get a similar level of confidence. I would say a test for length and a test for `objectContaining` on each element is enough